src/sail_json_backend/json.ml

```diff
@@ -231,20 +232,70 @@ let parse_encdec i mc format =
       end
   | _ -> assert false
 
+let filter_non_operands components =
+  let rec aux acc = function
+    | [] -> List.rev acc
+    | hd :: tl -> if String.trim hd = "spc" then tl else aux (hd :: acc) tl
+  in
+  aux [] components
```

Comment:
I took a stab at an alternative implementation that doesn't use `List.rev`:
```
let rec filter_non_operands components = match components with
  | [] -> []
  | hd :: tl when String.trim hd = "spc" -> tl
  | _ -> filter_non_operands (List.tl components)
```

What do you think? It's slightly smaller, doesn't require an embedded `aux` function, and doesn't use `List.rev`, so it could be slightly faster (although we're not terribly concerned with performance at this point).

I think it could be adapted to be used in `extract_operands`, below, as well.

Reply:
It's much more readable. I've adapted it for `extact_operands`